code/game/objects/items/rcd/RTD.dm

```diff
@@ -194,6 +197,11 @@
 
 	return data
 
+/obj/item/construction/rtd/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
+	playsound(src, SFX_TOOL_SWITCH, 20, TRUE)
```

Comment:
Move this to handle_ui_act(). Don't override this proc

Reply:
Why?

Reply:
Because it's overall less code. handle_ui_act() is called by ui_act() so there's no need to override it